kore/src/Kore/Step/EquationalStep.hs

```diff
@@ -384,14 +381,13 @@ applyRulesSequence
     -> [EqualityPattern variable]
     -- ^ Rewrite rules
     -> unifier (Results EqualityPattern variable)
-applyRulesSequence
-    sideCondition
-    initial
-    -- Wrap the rules so that unification prefers to substitute
-    -- axiom variables.
-    (map EqualityPattern.toAxiomVariables -> rules)
-  = matchRules sideCondition initial rules
+applyRulesSequence sideCondition initial rules =
+    matchRules sideCondition initial rules'
     >>= finalizeRulesSequence sideCondition initial
+  where
+    -- TODO (thomas.tuegel): Include freeVariables sideCondition
```

Comment:
This TODO raises an interesting question:

This is called for a term somewhere inside the configuration, or inside the main condition, or something like that. If we want to avoid the variables in the side condition, shouldn't we also avoid the variables in the main configuration, but outside the term we're simplifying?

(there's probably a simple explanation for that, but it's easier to just ask)

Reply:
I thought about this for a while. We do not need to avoid variables elsewhere in the configuration because:

1. There is no risk of confusing during rule application (matching, simplification, etc.) because the variable won't appear in this sub-term.
1. Any *new* variable introduced by the rule will be existentially quantified, so there is also no risk of confusion after the result is returned.